Core/include/Acts/Utilities/RangeXD.hpp

```diff
@@ -0,0 +1,228 @@
+  ///
+  /// One range is a superset of another range if and only if all points
+  /// contained within the second range are also contained within the first
+  /// range. Alternatively, this is equivalent to each of the one-dimensional
+  /// ranges in the first range being a superset of the corresponding
+  /// one-dimensional range in the second range.
+  ///
+  /// @param o The other range to compare to
+  ///
+  /// @return true The left-hand range is a superset of the right-hand range
+  /// @return false The left-hand range is not a superset of the right-hand
+  /// range
+  bool operator>=(const RangeXD<Dims, Type, Vector>& o) const {
+    for (std::size_t i = 0; i < Dims; ++i) {
+      if (!(this->m_dims[i] >= o[i])) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /// @brief Compute the intersection of this range with another range
+  ///
+  /// The intersection of one orthogonal range with another orthogonal range is
+  /// in itself an orthogonal range. This operation is commutative. This
+  /// intersection between two n-dimensional ranges is defined simply as the
+  /// intersection in each dimension of the two ranges.
+  ///
+  /// @param o The orthogonal range to compute the intersection with
+  ///
+  /// @return The intersection between the ranges
+  RangeXD<Dims, Type, Vector> operator&(
+      const RangeXD<Dims, Type, Vector>& o) const {
+    RangeXD<Dims, Type> res;
+
+    for (std::size_t i = 0; i < Dims; ++i) {
+      res[i] = m_dims[i] & o[i];
+    }
+
+    return res;
+  }
+
+  /// @brief Update the range to the intersection with another range
+  ///
+  /// This is the assignment version of the operator& method, meaning that it
+  /// updates the object on which it is called rather than producing a new
+  /// range.
+  ///
+  /// @param o The range to compute the intersection with
+  ///
+  /// @return This object
+  RangeXD<Dims, Type, Vector>& operator&=(
+      const RangeXD<Dims, Type, Vector>& o) {
+    for (std::size_t i = 0; i < Dims; ++i) {
+      m_dims[i] &= o[i];
+    }
+
+    return *this;
+  }
+
+  /// @brief Determine whether this range intersects another
+  ///
+  /// Two n-dimensional ranges intersect if and only if they intersect in every
+  /// one of their n dimensions. Otherwise, they are disjoint.
+  ///
+  /// @param r The other range to check
+  ///
+  /// @return true The ranges intersect
+  /// @return false The ranges do not intersect
+  bool operator&&(const RangeXD<Dims, Type, Vector>& r) const {
+    for (std::size_t i = 0; i < Dims; ++i) {
+      if (!(m_dims[i] && r[i])) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /// @brief Represent the range as a string
+  ///
+  /// This method produces a helpful string that can be used to debug the range
+  /// if needed. Not really designed to be used in production code.
+  ///
+  /// @return A string representing the range
+  std::string str(void) const {
```

Comment:
We have this at other places, can you please check if this is called `str(void)` or (what I have in mind) `toString(void)`?

I just want to make sure we keep similar naming conventions.

Reply:
Indeed it seems it's `toString(void)` in most places in the Acts code. I'll change it!